rest/oidc_api_test.go

```diff
@@ -1,139 +1,455 @@
 package rest
 
 import (
-	"io"
+	"encoding/json"
+	"fmt"
 	"net/http"
+	"net/http/cookiejar"
 	"net/http/httptest"
 	"net/url"
-	"strings"
 	"testing"
+	"time"
 
 	"github.com/couchbase/sync_gateway/auth"
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/gorilla/mux"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"gopkg.in/square/go-jose.v2/jwt"
 )
 
+var (
+	issuerGoogle      = ""
+	issuerFacebook    = ""
+	wantTokenResponse OIDCTokenResponse
+	invalidateState   bool
```

Comment:
We really should avoid any global variables unless there's a good reason for them, especially invalidateState which tests are changing to control behaviour of a shared function.

A safer approach to injecting test-specific behaviour than overriding these globals from within tests would be to take a parameter into `mockAuthServer` to control this invalidateState behaviour, and have each subtest start its own mockAuthServer with the desired value.

Reply:
Would like to discuss this approach once again before implementing.

Reply:
Removed global variables and added mockoptions map to keep the required mocking option in memory during test runs.